CesiumIonClient/src/Connection.cpp

```diff
@@ -373,6 +374,43 @@ Asset jsonToAsset(const rapidjson::Value& item) {
 
 } // namespace
 
+CesiumAsync::Future<std::optional<std::string>>
+CesiumIonClient::Connection::getApiUrl(
+    const CesiumAsync::AsyncSystem& asyncSystem,
+    const std::shared_ptr<IAssetAccessor>& pAssetAccessor,
+    const std::string& ionUrl) {
+  std::string configUrl = Uri::resolve(ionUrl, "config.json");
+  if (configUrl == "config.json") {
+    return asyncSystem.createResolvedFuture<std::optional<std::string>>(
+        std::nullopt);
+  }
+  return pAssetAccessor->get(asyncSystem, configUrl)
+      .thenImmediately([ionUrl](std::shared_ptr<IAssetRequest>&& pRequest) {
+        const IAssetResponse* pResponse = pRequest->response();
+        if (pResponse && pResponse->statusCode() >= 200 &&
+            pResponse->statusCode() < 300) {
+          rapidjson::Document d;
+          if (parseJsonObject(pResponse, d) && d.IsObject()) {
+            const auto itr = d.FindMember("apiHostname");
+            if (itr != d.MemberEnd() && itr->value.IsString()) {
+              return std::make_optional<std::string>(itr->value.GetString());
```

Comment:
I think this will be just a hostname, e.g., `api.cesium.com`, rather than a full URL, like `https://api.cesium.com`.